src/Forms/EditFormFactory.php

```diff
@@ -55,6 +57,33 @@ protected function getFormFields(RequestHandler $controller = null, $name, $cont
         return $fields;
     }
 
+    protected function getFormValidator(RequestHandler $controller = null, $name, $context = [])
```

Comment:
If we remove this method, the validation will still work, right? It'll just skip the JS validation and validate it on the backend only?

Seems like that would mean:
1. less complexity
2. better UX when there are validation errors in other form validators
3. only slightly worse (slower) UX when there are only required fields validation errors

If my assumptions above are correct, IMO we should remove this method.

Reply:
We shouldn't be removing JS validation, it's an existing feature

Reply:
No validation existing in elemental blocks before this, so javascript validation in elemental blocks was not an existing feature - and this is adding extra code to get it working. Which means as I stated above:
1. more complexity
2. worse UX in some scenarios
3. Only avoids a single fast XHR request

I can see you won't budge on this though so it will be part of the card I mentioned in https://github.com/silverstripe/silverstripe-elemental/pull/1150#issuecomment-2002616999

Reply:
JS validation is part of FormBuilder using FormSchema, it's not specific to elemental.

For instance try submitting a malformed email address in LinkField v4 email field - there's no XHR request because the validation happens in JS

It's pretty inconsistent which fields have it, for instance the new UrlField (external link field) has no JS validation and will get validated on the server - you'll see an XHR request for that one

Reply:
That's fine, like I said, I can see we won't agree about it so I'll leave it alone in this PR and include it in a separate card.